actors/test/replica_update_test.go

```diff
@@ -469,6 +469,218 @@ func TestExtendAfterUpdgrade(t *testing.T) {
 	assert.Equal(t, abi.ChainEpoch(miner.MaxSectorExpirationExtension - 1), infoFinal.Expiration-infoFinal.Activation)
 }
 
+func TestWrongDeadlineIndexFailure(t *testing.T) {
+	ctx := context.Background()
+	blkStore := ipld.NewBlockStoreInMemory()
+	v := vm.NewVMWithSingletons(ctx, t, blkStore)
+	addrs := vm.CreateAccounts(ctx, t, v, 1, big.Mul(big.NewInt(100_000), big.NewInt(1e18)), 93837778)
+
+	// create miner
+	sealProof := abi.RegisteredSealProof_StackedDrg32GiBV1_1
+
+	wPoStProof, err := sealProof.RegisteredWindowPoStProof()
+	require.NoError(t, err)
+	owner, worker := addrs[0], addrs[0]
+	minerAddrs := createMiner(t, v, owner, worker, wPoStProof, big.Mul(big.NewInt(10_000), vm.FIL))
+
+	// advance vm so we can have seal randomness epoch in the past
+	v, err = v.WithEpoch(abi.ChainEpoch(200))
+	require.NoError(t, err)
+
+	v, deadlineIndex, partitionIndex, sectorNumber := createSector(t, v, worker, minerAddrs.IDAddress, 100, sealProof)
+
+	// make some unverified deals
+	dealIDs := createDeals(t, 1, v, worker, worker, minerAddrs.IDAddress, sealProof)
+
+	// replicaUpdate the sector
+
+	replicaUpdate := miner.ReplicaUpdate{
+		SectorID:           sectorNumber,
+		Deadline:           deadlineIndex + 1,
+		Partition:          partitionIndex,
+		NewSealedSectorCID: tutil.MakeCID("replica1", &miner.SealedCIDPrefix),
+		Deals:              dealIDs,
+		UpdateProofType:    abi.RegisteredUpdateProof_StackedDrg32GiBV1,
+	}
+
+	ret := vm.ApplyCode(t, v, addrs[0], minerAddrs.RobustAddress, big.Zero(),
+		builtin.MethodsMiner.ProveReplicaUpdates,
+		&miner.ProveReplicaUpdatesParams{Updates: []miner.ReplicaUpdate{replicaUpdate}},
+		exitcode.ErrIllegalArgument)
+
+	_, ok := ret.(*bitfield.BitField)
+	require.False(t, ok)
+
+	newSectorInfo := vm.SectorInfo(t, v, minerAddrs.RobustAddress, sectorNumber)
```

Comment:
Nicer to assert that the "new" sectorinfo is exactly equal to the "old" sectorinfo, i think...you can just require.Equal the structs themselves